mkdocs_redirects/plugin.py

```diff
@@ -125,12 +130,13 @@ def on_post_build(self, config, **kwargs):
 
         # Walk through the redirect map and write their HTML files
         for page_old, page_new in self.redirects.items():
+            new_path_hash_free, _ = _split_hash_fragment(str(page_new))
```

Comment:
Please let's not silently drop any part. This is just confusing. Kinda unrelated change, anyway.

Reply:
All the other suggestions make perfect sense to me and work great, and I've incorporated them locally. Thank you.

Without this change (and the associated change on line 139), I get a warning about non-existent target pages when building documentation using `mkdocs serve` and an HTTP 404 error when navigating to the source URL in the browser. The warnings and errors are exclusive to targets with hash fragments.

With this change there is no warning and the source URL correctly redirects to the target.

I'm open to suggestions on improving clarity, or a better place in the code to make this change.

Reply:
Ah yes, this suggestion doesn't make sense. I thought it's changing page_old but it's changing page_new - that's all good.